tests/python_package_test/test_sklearn.py

```diff
@@ -1290,6 +1290,90 @@ def test_max_depth_warning_is_never_raised(capsys, estimator_class, max_depth):
     assert "Provided parameters constrain tree depth" not in capsys.readouterr().out
 
 
+def test_getting_feature_names_in_np_input():
```

Comment:
There is quite a lot of repetition in these tests. Generally I'm supportive of repetition in tests in favor of making it easier to diagnose issues or selectively include / exclude tests... but for the cases of `.feature_names_in_` and `.get_feature_names_out()`, I think the tests should be combined.

So can you please reduce these 4 tests down to 2? One for `numpy` input without feature names, one for `pandas` input with feature names?

Ending with assertions like this:

```python
expected_col_names = np.array([f"Column_{i}" for i in range(X.shape[1])]
np.testing.assert_array_equal(model.feature_names_in_, expected_col_names)
np.testing.assert_array_equal(model.get_feature_names_out(), expected_col_names)
```